src/Domain/Creators/TimeDependentOptions/BinaryCompactObject.cpp

```diff
@@ -71,11 +72,91 @@ TimeDependentMapOptions<IsCylindrical>::TimeDependentMapOptions(
+               {gsl::at(expansion_map_options_.value().initial_values, 1)}}},
+          initial_expiration_time, false);
+  result[expansion_outer_boundary_name] =
+      std::make_unique<FunctionsOfTime::FixedSpeedCubic>(
+          1.0, initial_time_,
+          expansion_map_options_.value().outer_boundary_velocity,
+          expansion_map_options_.value().outer_boundary_decay_time);
+  ASSERT(rotation_map_options_.has_value(),
+         "Initial values for the rotation map need to be provided when using "
+         "the worldtube.");
+  result[rotation_name] =
+      std::make_unique<FunctionsOfTime::IntegratedFunctionOfTime>(
+          initial_time_,
+          std::array<double, 2>{
+              0.,
+              gsl::at(rotation_map_options_.value().initial_angular_velocity,
+                      2)},
+          initial_expiration_time, true);
+
+  // Size and Shape FunctionOfTime for objects A and B. Only spherical excision
+  // spheres are supported currently.
+  ASSERT(
+      shape_options_A_.has_value(),
+      "Initial size for both excision spheres need to be provided when using "
+      "the worldtube.");
+  ASSERT(
+      shape_options_B_.has_value(),
+      "Initial size for both excision spheres need to be provided when using "
+      "the worldtube.");
+  for (size_t i = 0; i < shape_names.size(); i++) {
+    const auto make_initial_size_values = [](const auto& lambda_options) {
+      return std::array<double, 2>{
+          {gsl::at(lambda_options.value().initial_size_values.value(), 0),
+           gsl::at(lambda_options.value().initial_size_values.value(), 1)}};
+    };
+    const std::array<double, 2> initial_size_values =
+        i == 0 ? make_initial_size_values(shape_options_A_)
+               : make_initial_size_values(shape_options_B_);
+    const size_t initial_l_max = 2;
+    const DataVector shape_zeros{
+        ylm::Spherepack::spectral_size(initial_l_max, initial_l_max), 0.0};
+
+    result[gsl::at(shape_names, i)] =
+        std::make_unique<FunctionsOfTime::PiecewisePolynomial<2>>(
+            initial_time_,
+            std::array<DataVector, 3>{shape_zeros, shape_zeros, shape_zeros},
+            std::numeric_limits<double>::infinity());
+    result[gsl::at(size_names, i)] =
+        std::make_unique<FunctionsOfTime::IntegratedFunctionOfTime>(
+            initial_time_, initial_size_values, initial_expiration_time, false);
+  }
+  return result;
+}
+
+template <bool IsCylindrical>
+template <bool UseWorldtube>
 std::unordered_map<std::string,
                    std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>
 TimeDependentMapOptions<IsCylindrical>::create_functions_of_time(
     const std::unordered_map<std::string, double>& initial_expiration_times)
     const {
+  if constexpr (UseWorldtube) {
+    static_assert(not IsCylindrical,
+                  "Cylindrical map not supported with worldtube");
+    return create_worldtube_functions_of_time();
```

Comment:
Also `(void)initial_expiration_times`. Or you could optionally check that the initial expiration times are empty because there shouldn't be any control systems for the CSW system.